src/model/cfg.py

```diff
@@ -20,25 +23,120 @@ class DatasetType(Enum):
         "/media/omarabdelgawad/New Volume/Datasets/image_coloring/natural_view/"
     )
 
+# custom parser
+def positive_float(value: str) -> float:
+    ivalue = float(value)
+    if ivalue <= 0:
+        raise argparse.ArgumentTypeError(f"{value} is an invalid positive float value")
+    return ivalue
 
+def positive_int(value) -> int:
+    ivalue = int(value)
+    if ivalue <= 0:
+        raise argparse.ArgumentTypeError(f"{value} is an invalid positive int value")
+    return ivalue
+
+def custom_arg_parser(argv: Optional[Sequence[str]] = None) -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="An unsupervised image-to-image translation model",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-lr", "--learning_rate",
+        type=positive_float,
+        default=2e-4,
+        help="Learning rate",
+        metavar="LEARNING_RATE",
+        dest="lr",
+    )
+    parser.add_argument(
+        "-b", "--batch_size",
+        type=positive_int,
+        default=16,
+        help="Batch size",
+        metavar="BATCH_SIZE",
+        dest="b",
+    )
+    parser.add_argument(
+        "-w", "--num_workers",
+        type=int,
+        default=2,
+        choices=[1, 2, 3, 4, 5, 6, 7, 8],
+        help="Number of workers",
+        metavar="NUM_WORKERS",
+        dest="w",
+    )
+    parser.add_argument(
+        "-i", "--image_size",
+        type=int,
+        default=256,
+        choices=[256, 512, 1024],
+        help="Image size",
+        metavar="IMAGE_SIZE",
+        dest="i",
+    )
+    parser.add_argument(
+        "-e", "--num_epochs",
+        type=positive_int,
+        default=100,
+        help="Number of epochs",
+        metavar="NUM_EPOCHS",
+        dest="e",
+    )
+    parser.add_argument(
+        "-l", "--load_model",
+        action="store_false",
+        default=True,
+        help="Load model",
+        dest="l",
+    )
+    parser.add_argument(
+        "-s", "--save_model",
+        action="store_true",
+        default=False,
+        help="Save model",
+        dest="s",
+    )
+    parser.add_argument(
+        "-d", "--num_images_dataset",
+        type=positive_int,
+        default=1000,
+        help="Number of images in dataset",
+        metavar="NUM_IMAGES_DATASET",
+        dest="d",
+    )
+    parser.add_argument(
+        "-v", "--val_batch_size",
+        type=positive_int,
+        default=8,
+        help="Validation batch size",
+        metavar="VAL_BATCH_SIZE",
+        dest="v",
+    )
+
+    args = parser.parse_args(argv)
+    return args
+
+args = custom_arg_parser()
 DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
-LEARNING_RATE = 2e-4
-BATCH_SIZE = 16
-NUM_WORKERS = 2
-IMAGE_SIZE = 256
+LEARNING_RATE = args.lr
+BATCH_SIZE = args.b
+NUM_WORKERS = args.w
+IMAGE_SIZE = args.i
```

Comment:
Please change all dest attributes to be more descriptive. I recommend that you use the long option instead for consistency excep with special cases such as `SAVE_MODEL`. Also note that the convention for long command line arguments is to use hyphens `-` instead of underscores `_`. So please update all long options accordingly.